extensions/kill-process/src/index.tsx

```diff
@@ -61,25 +80,99 @@ export default function ProcessList() {
     exec(`kill -9 ${process.id}`);
     setState(state.filter((p) => p.id !== process.id));
     clearSearchBar({ forceScrollToTop: true });
-    showHUD(`✅ Killed ${process.name === "-" ? `process ${process.id}` : process.name}`);
-  };
-
-  const copyToClipboardAction = (process: Process) => {
-    return process.path == null ? null : <Action.CopyToClipboard title="Copy Path" content={process.path} />;
+    showHUD(`✅ Killed ${process.processName === "-" ? `process ${process.id}` : process.processName}`);
   };
 
   const subtitleString = (process: Process) => {
-    let subtitle = undefined;
-
+    const subtitles = [];
+    if (process.type == "aggregatedApp" && process.appName != undefined) {
+      subtitles.push(process.appName);
+    }
     if (shouldShowPID) {
-      subtitle = process.id;
+      subtitles.push(process.id.toString());
     }
-
     if (shouldShowPath) {
-      subtitle = subtitle ? `${subtitle} - ${process.path}` : process.path;
+      subtitles.push(process.path);
     }
+    return subtitles.join(" - ");
+  };
 
-    return subtitle;
+  const aggregate = (processes: Process[]): Process[] => {
+    const result = Array<Process>();
+    type ProcessNode = {
+      process: Process | undefined;
+      childNodes: ProcessNode[];
+    };
+    const appMap = new Map<number, ProcessNode>();
```

Comment:
nit: maybe lodash groupby might be helpful?

Reply:
I'll get to know that

Reply:
It seems `groupby` will have less help to the aggregating algorithm, because it will not only aggregate by parent id, but also grandparent and grand-grandparent id... It's recursive and using `groupby` seems will have much more complexity with less benefit.